logger/style_guide.md

```diff
@@ -0,0 +1,157 @@
+# Logging Style Guide
+
+The intention of logging is to give insight to the administrator of how
+the server is running and also notify the administrator of any problems
+or potential problems with the system.
+
+Great care should be taken when adding a logging message. There is a
+line between too much information and too little information that is
+very difficult to understand and even more difficult to reach.
+
+InfluxDB uses structured logging. Structured logging is when you log
+messages and attach context to those messages with more easily read data
+regarding the state of the system. A structured log message is composed
+of a time, a level, a message, and additional context about the message
+itself.
+
+**Log messages** should be simple statements or phrases that begin with
+a capital letter, but have no period at the end. The message should be a
+constant so that every time it is logged it is easily identified and can
+be filtered by without regular expressions.
+
+Any **dynamic content** should be expressed by context. The key should
+be a constant and the value is the dynamic content.
+
+Do not log messages in tight loops or other high performance locations.
+It will likely create a performance problem.
+
+## Naming Conventions
+
+If the log encoding format uses keys for the time, message, or level,
+the key names should be `ts` for time, `msg` for the message, and
+`level` for the level.
+
+If the log encoding format does not use keys for the time, message, or
+level and instead outputs them in some other method, this guideline can
+be ignored. The output formats logfmt and json both use keys when
+encoding these values.
+
+### Context Key Names
+
+The key for the dynamic content in the context should be formatted in
+camelCase. For acronyms, each letter of the acronym should match the
```

Comment:
I personally like using dashes, but I think last time I brought this up, someone requested `camelCase`. We should just put it to a vote. I'm not a big fan of snake case mostly because I think it looks ugly in output, but whichever one a plurality wants, I'll take.